### PR TITLE
fix #22 - update min req & boost CI suite #20

### LINKED ISSUE #22: [COMPAT] `fastpairs` not compatible with `scipy>=1.12`
Currently `fastpairs` is not compatible with `scipy>=1.12` – [here](https://github.com/carsonfarmer/fastpair/blob/d3170fd7e4d6e95312e7e1cb02e84077a3f06379/fastpair/test/test_fastpair.py#L23)

```
fastpair/test/test_fastpair.py:23: in <module>
    from scipy import mean, array, unique
ImportError: cannot import name 'mean' from 'scipy'
```

xref #20

### PR DESCRIPTION
**Pull request by jGaboardi**
This PR:
* resolves #22 
   * swaps put using old `scipy` methods for `numpy` ones
* set minimum required version from [SPEC000](https://scientific-python.org/specs/spec-0000/)
* fill out CI suite with oldest Python 3.10 to bleeding edge Python 3.12
   * test latest Python 3.12 against intel macOS & arm macOS + windows 
* xref #20

### CODE REVIEW

File: ci/py311-latest.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+name: py311-latest
+channels:
+  - conda-forge
+dependencies:
+  - python=3.12
```
Comment (carsonfarmer) [inline ci/py311-latest.yaml:5]: should this be 3.11?

**Review by carsonfarmer — APPROVED** [review]

File: ci/py311-latest.yaml (hunk 1 above)
Comment (jGaboardi) [reply ci/py311-latest.yaml:5]: Yes! Great catch.

Comment (jGaboardi) [inline ci/py311-latest.yaml:5]:
```suggestion
  - python=3.11
```

**Comment by jGaboardi** [comment]
Failure due to fluke in `codecov` action. Proceeding with merge.

https://github.com/carsonfarmer/fastpair/actions/runs/9455680494/job/26045980185?pr=33#step:7:1

**Pull request merged**